tests/infer/test_advi.py

```diff
@@ -99,3 +99,27 @@ def model():
     assert quantiles["z"][0] + 0.1 < quantiles["z"][1]
     assert quantiles["z"][1] + 0.1 < quantiles["z"][2]
     assert quantiles["z"][2] < 0.99
+
+
+@pytest.mark.parametrize("continuous_class", [ADVIMultivariateNormal, ADVIDiagonalNormal])
+def test_discrete_parallel(continuous_class):
+    K = 2
+    data = torch.tensor([0., 1., 10., 11., 12.])
+
+    def model(data):
+        weights = pyro.sample('weights', dist.Dirichlet(0.5 * torch.ones(K)))
+        locs = pyro.sample('locs', dist.Normal(0, 10).reshape([K]))
+        scale = pyro.sample('scale', dist.LogNormal(0, 1))
+
+        with pyro.iarange('data'):
```

Comment:
This iarange ends up allocating multiple dimensions in the global `_DIM_ALLOCATOR` when the `ADVIMaster` and `ADVIDiscreteParallel` both create [differently-blocked] `prototype_trace`s.